src/hooks/useKanban.js

```diff
@@ -43,7 +43,22 @@ export function useKanban() {
     const unsubscribeColumns = onSnapshot(q, (snapshot) => {
       const newColumns = {};
       snapshot.forEach((doc) => {
-        newColumns[doc.id] = { id: doc.id, ...doc.data() };
+        const columnId = String(doc.id);
+        const columnData = doc.data();
+        
+        // Ensure tasks is always an array and all IDs are strings
+        const tasks = Array.isArray(columnData.tasks) 
+          ? columnData.tasks.map(task => ({
+              ...task,
+              id: String(task.id || crypto.randomUUID())
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
id: String(task.id || ((() => {
  try {
    return crypto.randomUUID();
  } catch (e) {
    return Date.now().toString(36) + Math.random().toString(36).substr(2);
  }
})()))
```
The crypto.randomUUID() usage for generating IDs should be wrapped in a try-catch block as it might not be available in all browsers. Consider providing a fallback mechanism.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;